truffle/src/main/java/org/jruby/truffle/nodes/literal/StringLiteralNode.java

```diff
@@ -9,41 +9,33 @@
  */
 package org.jruby.truffle.nodes.literal;
 
-import com.oracle.truffle.api.CompilerDirectives.CompilationFinal;
 import com.oracle.truffle.api.frame.VirtualFrame;
 import com.oracle.truffle.api.object.DynamicObject;
 import com.oracle.truffle.api.source.SourceSection;
-import org.jcodings.Encoding;
 import org.jruby.truffle.nodes.RubyNode;
 import org.jruby.truffle.nodes.objects.AllocateObjectNode;
 import org.jruby.truffle.nodes.objects.AllocateObjectNodeGen;
 import org.jruby.truffle.runtime.RubyContext;
+import org.jruby.truffle.runtime.rope.LeafRope;
+import org.jruby.truffle.runtime.rope.RopeOperations;
 import org.jruby.util.ByteList;
 
-import java.util.Arrays;
-
 public class StringLiteralNode extends RubyNode {
 
-    private @CompilationFinal byte[] bytes; // deeply immutable
-    private final Encoding encoding;
-    private final int codeRange;
+    private final LeafRope rope;
 
     @Child private AllocateObjectNode allocateObjectNode;
 
     public StringLiteralNode(RubyContext context, SourceSection sourceSection, ByteList byteList, int codeRange) {
         super(context, sourceSection);
         assert byteList != null;
-        this.bytes = byteList.bytes();
-        this.encoding = byteList.getEncoding();
-        this.codeRange = codeRange;
+        this.rope = RopeOperations.create(byteList.bytes(), byteList.getEncoding(), codeRange);
```

Comment:
We should de-dup here or in the caller so that literals in different places by the same bytes will be the same `LeafRope` instance.